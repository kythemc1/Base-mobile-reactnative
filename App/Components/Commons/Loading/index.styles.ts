import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
    mContainer: {
        position: 'absolute',
        backgroundColor: 'rgba(0, 0, 0, 0.4)',
        width: '100%',
        height: '100%',
        justifyContent: 'center',
        alignItems: 'center',
        zIndex: 1
    }
});
