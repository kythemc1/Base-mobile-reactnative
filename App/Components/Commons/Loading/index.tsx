import React from 'react';
import { ActivityIndicator, View } from 'react-native';
import { colors } from 'Configs/UI/Colors';
import { styles } from 'Components/Commons/Loading/index.styles';

interface State {
  loading: boolean | undefined;
}

export default function Loading({ loading }: State) {
    return (
        loading ?
            <View style={styles.mContainer}>
                <ActivityIndicator animating={loading} size={'large'} color={colors.color_button_confirm} />
            </View> : null
    );
}
