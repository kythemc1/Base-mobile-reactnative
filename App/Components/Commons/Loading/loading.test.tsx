import renderer from 'react-test-renderer';
import Loading from 'Components/Commons/Loading/index';
jest.useRealTimers();
test('Test loading component', () => {
    const tree = renderer.create(<Loading loading={true} />).toJSON();
    expect(tree).toMatchSnapshot();
});
