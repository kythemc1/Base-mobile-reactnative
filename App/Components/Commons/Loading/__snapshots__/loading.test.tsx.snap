// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test loading component 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "rgba(0, 0, 0, 0.4)",
      "height": "100%",
      "justifyContent": "center",
      "position": "absolute",
      "width": "100%",
      "zIndex": 1,
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#143980"
    size="large"
  />
</View>
`;
